File: WRBlueScreen/WRBlueScreen.m
#import "WRBlueScreen.h"
#import "WRBlueScreenView.h"

@interface WRBlueScreen()

@property (nonatomic, weak) WRBlueScreenView *blueScreenView;

@end

@implementation WRBlueScreen

- (instancetype)init {
    self = [super initWithFrame:[UIScreen mainScreen].bounds];
    if (self) {
        self.windowLevel = UIWindowLevelStatusBar;
        [self addSubview:self.blueScreenView];
        self.shouldCrashApplication = true;
        self.percentageChanceToCrash = 70;
        self.minIntervalToCrash = 2;
        self.maxIntervalToCrash = 5;
    }
    return self;
}

+ (instancetype)sharedWindow {
    static WRBlueScreen *_sharedInstance = nil;
    static dispatch_once_t oncePredicate;
    dispatch_once(&oncePredicate, ^{
        _sharedInstance = [[WRBlueScreen alloc] init];
    });
    return _sharedInstance;
}

- (WRBlueScreenView *)blueScreenView {
    if (!_blueScreenView) {
        _blueScreenView = [WRBlueScreenView fromNibFile];
        _blueScreenView.frame = self.bounds;
    }
    return _blueScreenView;
}

- (void)showErrorMessage:(NSString *)errorMessage {
    [self hideKeybaord];
    [self.blueScreenView showErrorMessage:errorMessage];
    self.hidden = NO;
    [self makeKeyAndVisible];
    if (self.shouldCrashApplication) {
        [self randomCrashApp];
    }
}

- (void)showError:(NSError *)error {
    NSString *errorMessage = [NSString stringWithFormat:@"%ld : %p", (long)error.code, error];
    [self showErrorMessage:errorMessage];
}

- (void)hideKeybaord {
    [UIView beginAnimations:nil context:nil];
    [UIView setAnimationDuration:0];
    [UIView setAnimationDelay:0];
    [UIView setAnimationCurve:UIViewAnimationCurveLinear];
    for (UIWindow *window in [UIApplication sharedApplication].windows) {
        [window endEditing:YES];
    }
    [UIView commitAnimations];
}

- (void)randomCrashApp {
    NSInteger randomNumber = 1 + arc4random() % 100;
    if (randomNumber <= self.percentageChanceToCrash) {
        NSInteger randomCrashTime = 0;
        if (self.minIntervalToCrash >= self.maxIntervalToCrash) {
            randomCrashTime = self.minIntervalToCrash;
        } else {
            randomCrashTime = self.minIntervalToCrash + arc4random() % (self.maxIntervalToCrash - self.minIntervalToCrash + 1);
        }
        [NSTimer scheduledTimerWithTimeInterval:randomCrashTime target:self selector:@selector(crashApp) userInfo:nil repeats:FALSE];
    }
}

- (void)crashApp {
    NSLog(@"BSOD");
    [@[] objectAtIndex:666];
}

@end

File: WRBlueScreen/WRBlueScreen.swift
import Foundation
import Darwin

public class WRBlueScreen: UIWindow {
    
    public static let sharedWindow = WRBlueScreen()
    public var shouldCrashApplication = true
    public var percentageChanceToCrash:Int = 70
    public var minIntervalToCrash:Int = 2
    public var maxIntervalToCrash:Int = 5
    
    private lazy var blueScreenView:WRBlueScreenView = WRBlueScreenView.fromNibFile()

    init() {
        super.init(frame: UIScreen.mainScreen().bounds)
        blueScreenView.frame = self.bounds
        self.addSubview(blueScreenView)
    }

    required public init?(coder aDecoder: NSCoder) { super.init(coder: aDecoder) }
    
    public func showErrorMessage(errorMessage:String) {
        hideKeybaord()
        blueScreenView.showErrorMessage(errorMessage)
        hidden = false
        makeKeyAndVisible()
        if shouldCrashApplication { randomCrashApp() }
    }
    
    public func showError(error: NSError) {
        self.showErrorMessage("\(error.code) : \(unsafeAddressOf(error))")
    }
    
    private func hideKeybaord() {
        UIView.beginAnimations(nil, context: nil)
        UIView.setAnimationDuration(0.0)
        UIView.setAnimationDelay(0.0)
        UIView.setAnimationCurve(.Linear)
        _ = UIApplication.sharedApplication().windows.map { $0.endEditing(true) }
        UIView.commitAnimations()
    }
        
    private func randomCrashApp() {
        let randomNumber = 1 + arc4random() % 100
        if randomNumber <= UInt32(percentageChanceToCrash) {
            let randomCrashTime =
                minIntervalToCrash >= maxIntervalToCrash ? minIntervalToCrash : minIntervalToCrash + Int(arc4random()) % (maxIntervalToCrash - minIntervalToCrash + 1)
            NSTimer.scheduledTimerWithTimeInterval(Double(randomCrashTime), target: self, selector: Selector("crashApp"), userInfo: nil, repeats: false)
        }
    }
    
    func crashApp() {
        print("BSOD")
        [][666]
    }
}